src/reducer.js

```diff
@@ -0,0 +1,22 @@
+// @flow
+
+import type { NavigationAction, NavigationState } from 'react-navigation';
+import type { Reducer } from 'redux';
+import type { Navigator, ReducerState } from './types'
+
+import { NavigationActions } from "react-navigation";
+
+export function createNavigationReducer(navigator: Navigator): Reducer<*, *> {
+  const initialState = navigator.router.getStateForAction(
+    NavigationActions.init(),
+    null
+  );
+
+  return (
+    state: ReducerState = initialState,
+    action: NavigationAction
+  ): ReducerState => {
+    const newState = navigator.router.getStateForAction(action, state);
```

Comment:
I looked into this a bit. I think we should be returning whatever the reducer (`getStateForAction`) tells us to return, even if it is falsey. This is the default `react-navigation` behavior when using a "stateful" navigation container, ie. without passing in the `state` prop). It's also what one would expect from a reducer.